index.js

```diff
@@ -173,52 +184,24 @@ function update(event) {
     console.log(hashstr);
     history.pushState(null, null,
                       '#' + LZString.compressToEncodedURIComponent(hashstr));
-    
-    var arc = {};
-    for (var i = 1; i <= 5; ++i) {
-        for (var c = 0; c < 8; ++c) {
-            if ($(colors[c] + '.' + i).checked) {
-                if (!(i in arc)) {
-                    arc[i] = [colors[c]];
-                } else {
-                    arc[i].push(colors[c]);
-                }
-            }
-        }
-    }
-    var major = [];
-    if ($('anytime').checked) {
-        for (var i = 1; i <= 2; ++i) {
-            major.push($('at' + i).value);
-        }
-    } else {
-        for (var i = 1; i <= 6; ++i) {
-            if ($('major' + i).value) {
-                major.push($('major' + i).value);
-            }
-        }
-    }
-    var flavor = [];
-    for (var i = 1; i <= 12; ++i) {
-        if ($('flavor' + i).value) {
-            flavor.push([$('cf'+i+'a').value.toLowerCase(), $('cf'+i+'b').value.toLowerCase(), $('flavor' + i).value]);
-        }
-    }
-    var html = cardHtml($('name').value, $('xp').value, arc, $('color').value.toLowerCase(), $('anytime').checked, major, flavor, $('desc').value);
-    let discord = cardDiscord($('name').value, $('xp').value, arc, $('color').value.toLowerCase(), $('anytime').checked, major, flavor, $('desc').value);
+  
+	translateFormToObject();
```

Comment:
What would you think of, instead of having a global quest, having translateFormToObject() return an object you then pass to cardHtml() and cardDiscord()? And then similarly translateObjectToForm() would take a parameter? I think that'd be easier to follow rather than worrying about what invariants apply to the global object and when it may or may not be up-to-date.

Reply:
Sure, that makes sense to me.